Update release notes for #67886 (#68224)
elastic/elasticsearch: docs/reference/release-notes/7.11.asciidoc

Mapping::
* Distinguish between simple matches with and without the terms index {es-pull}63945[#63945]

Templates::

> * Composable index templates with data streams can no longer contain alias definitions.
>   This also applies for any component templates that is referred by composable index templates
>   that contain data streams. The reason why this was allowed in ES 7.9 and 7.10 is a bug. {es-pull}67886[#67886]

Reviewer: I tried to streamline this a bit, but the original text is fine as is.
```suggestion
* Index templates with data stream enabled can no longer contain index aliases
or component templates with index aliases. A bug in {es} 7.10 and 7.9 previously
allowed this. {es-pull}67886[#67886]
```
